External/Plugins/BookmarkPanel/PluginUI.cs

```diff
@@ -228,20 +228,22 @@ private void InitializeTexts()
         /// </summary>
         private void InitializeGraphics()
         {
-            this.imageList = new ImageList();
+            this.imageList = new ImageListManager();
             this.imageList.ImageSize = ScaleHelper.Scale(new Size(16, 16));
             this.imageList.ColorDepth = ColorDepth.Depth32Bit;
-            this.imageList.Images.Add(PluginBase.MainForm.FindImage("559|26|0|1"));
-            this.imageList.Images.Add(PluginBase.MainForm.FindImage("229"));
-            this.imageList.Images.Add(PluginBase.MainForm.FindImage("197"));
-            this.imageList.Images.SetKeyName(0, "Bookmark");
-            this.imageList.Images.SetKeyName(1, "Info");
-            this.imageList.Images.SetKeyName(2, "Error");
+            this.imageList.Initialize(ImageList_OnInitialize);
             this.listView.SmallImageList = this.imageList;
             this.removeBookmarksItem.Image = PluginBase.MainForm.FindImage("402|4|4|4");
             this.searchButton.Image = PluginBase.MainForm.FindImage("484|26|-4|4");
         }
 
+        private void ImageList_OnInitialize(object sender, EventArgs e)
+        {
+            this.imageList.Images.Add("Bookmark", PluginBase.MainForm.FindImageAndSetAdjust("559|26|0|1"));
+            this.imageList.Images.Add("Info", PluginBase.MainForm.FindImageAndSetAdjust("229"));
+            this.imageList.Images.Add("Error", PluginBase.MainForm.FindImageAndSetAdjust("197"));
+        }
```

Comment:
Use `FindImageAndSetAdjust()` inside `ImageListManager.OnInitialize` event. It is equivalent to `ImageSetAdjust(FindImage(data, false))`, making sure that the image color is adjusted according to the theme but the image is not cached to be updated on theme change. (The cached image uses `WeakReference` however.)